From 1f54397a747bd9f9c1624fa568d0165daccbda52 Mon Sep 17 00:00:00 2001
From: ariel aviad <aaviad@habana.ai>
Date: Mon, 02 Sep 2024 10:02:15 +0300
Subject: [PATCH] [FSW-4085] boot info ver: rm support ver 0

Remove not supported version 0 in boot info
version 0 use sha256 that need to be removed

Change-Id: I7b054e956d30a570147856ebf50a57e73bce3b97
Signed-off-by: ariel aviad <aaviad@habana.ai>
---

diff --git a/lib/common/boot_info.c b/lib/common/boot_info.c
index 8f0975a..eaa26a1 100644
--- a/lib/common/boot_info.c
+++ b/lib/common/boot_info.c
@@ -9,7 +9,6 @@
 #include <baremetal.h>
 #include <hl_boot_desc.h>
 #include <common.h>
-#include <sha256.h>
 #include <crc32.h>
 
 #include "boot_info.h"
@@ -43,14 +42,9 @@
 			    struct hl_boot_info_desc *desc)
 {
 	uint32_t crc32_res;
-	uint16_t boot_info_version = hl_get_boot_info_ver();
 
-	if (!boot_info_version) {
-		sha256_csum(input, ilen, desc->chksum);
-	} else {
-		crc32_res = crc32(0, input, ilen);
-		memcpy(&desc->crc32, &crc32_res, sizeof(crc32_res));
-	}
+	crc32_res = crc32(0, input, ilen);
+	memcpy(&desc->crc32, &crc32_res, sizeof(crc32_res));
 }
 
 static int validate_sha_csum(struct hl_boot_info_ptrs *ptrs,
@@ -58,12 +52,11 @@
 			     unsigned int ilen,
 			     struct hl_boot_info_desc *desc)
 {
-	u8 chksum[HL_BOOT_INFO_CSUM_SIZE];
 	u32 crc32_res;
 
 	if (!ptrs->desc_ver) {
-		sha256_csum(input, ilen, chksum);
-		return memcmp(chksum, desc->chksum, HL_BOOT_INFO_CSUM_SIZE);
+		preboot_puts("boot info descriptor version 0 is no longer supported\n");
+		return -1;
 	}
 
 	/* for version >= 1 */
